src/System.Collections.Immutable/tests/ImmutableArrayTest.cs

```diff
@@ -251,7 +251,8 @@ public void Covariance()
             ImmutableArray<string> derivedImmutable = ImmutableArray.Create("a", "b", "c");
             ImmutableArray<object> baseImmutable = derivedImmutable.As<object>();
             Assert.False(baseImmutable.IsDefault);
-            Assert.Equal(derivedImmutable, baseImmutable);
+            // Must cast to object or the IEnumerable<object> overload of Equals would be used
+            Assert.Equal((object)derivedImmutable, baseImmutable);
```

Comment:
Seems you are correct. I was using this during my tests because the first thing I tried was this:

```csharp
Assert.Equal(derivedImmutable, baseImmutable, EqualityComparer<object>.Default);
```

This did not resolve the problem (it resolves to `Equal<T>(IEnumerable<T>, IEnumerable<T>, IEqualityComparer<T>`). I didn't realize that both the cast and the explicit comparer were the reason it started to work and removed it. Back now though. :smile: